buildLogic/toolchain-convention/src/main/kotlin/app/cash/sqldelight/toolchain/ToolchainConventions.kt

```diff
@@ -44,4 +44,4 @@ abstract class ToolchainConventions(private val targetJdkVersion: String) : Plug
 class RuntimeToolchainConventions : ToolchainConventions("1.8")
 
 // Controls the minimum JDK version required to run the SQLDelight plugin and compiler
-class CompilerToolchainConventions : ToolchainConventions("11")
+class CompilerToolchainConventions : ToolchainConventions("17")
```

Comment:
will this mean anyone using SQLDelight also has to build their project using JDK 17? Is there any way to avoid that?

Reply:
It is the same like AGP 8 but I think, it should be possible to avoid it using Gradle Workers, will try it.

Reply:
if AGP 8 requires 17 I'm fine with this

Reply:
Ohhh, while we can use the `JavaExecSpec` to specify the jvm version, it needs a Main class and passes all properties as string because it forks a new process, so it basically needs a cli version of sqldelight... This is also on our roadmap, but I definitely won't be able to finish it today.

Reply:
:+1: sounds good, lets merge as is then and we can revisit after we've pulled out the CLI compatible compiler api

Reply:
Sad to hear that 2.0.2 version dropped Java11 compatibility, this minor version brings too many uncertain changes to the existing huge project that is still using AGP7 (due to the removal of Transform in AGP8) and Java11. I don't know why the bytecode compatibility with Java11 must be dropped here, which will cause this project to not get important bug fixes of 2.x versions.